extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -335,6 +378,8 @@ public void stop(boolean stopGracefully)
         }
         log.info("Shutdown notice handled");
 
+        taskClient.close();
+        workerExec.shutdownNow();
```

Comment:
@pjain1 have you seen this issue while shutting down a supervisor? I considered this possibility but I think that in most cases the shutdown logic should take less than 15 seconds unless the tasks are for some reason not very responsive (and there was an issue around this but #3401 should have fixed it)